packages/headless/src/controllers/commerce/product-view/headless-product-view.ssr.ts

```diff
@@ -0,0 +1,39 @@
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
```

Comment:
Not a sub-controller, but this one had kind of slipped between the cracks, so I gave it a shot here.

This could (and perhaps should) actually be a util instead. It's not really a controller, because it doesn't have a state or a subscribe method (which is why I had to redefine a buildSSRProductView function here whose return value extends the Controller interface).

The thing I like about having it as a controller is that it makes it simple to use because it's consistent with everything else.

On the other hand, it's a bit awkward because it forces us to turn it into a "fake" controller. Also, specifying a solution type for this one feels odd... a product description page (in which the product view would be used) does not necessarily leverage any of those use cases. The only use of the product view "controller" is to log an `ec_productView` event through the Headless commerce Relay instance, and grabbing the necessary context information from the Headless commerce state when doing so.

Thoughts?

Reply:
I think the best solution is to keep it like this. It could be useful to add a sentence in the jsdoc explaining how this controller has no state.